src/components/Container.js

```diff
@@ -5,172 +5,223 @@ import Projection from 'ol/proj';
 import MapControls from './MapControls';
 import OpenLayersMap from './OpenLayersMap';
 import TrailList from './TrailList';
-import {getElevation} from '../utils/mapUtils';
-import kill_logo from './../imgs/Kill_Logo.png'
-import {Image} from 'react-bootstrap';
+import {getElevation, getMapStyle} from '../utils/mapUtils';
+import kill_logo from './../imgs/Kill_Logo.png';
+import { Grid } from 'material-ui';
+import { Image } from 'react-bootstrap';
+import { Trail, Hydrant } from '../utils/records';
 import ImportExport from './ImportExport';
 
-class Container extends React.Component{
-  constructor(props){
+class Container extends React.Component {
+  constructor(props) {
     super(props);
-    this.endDraw = this.endDraw.bind(this);
+    /*
+    const savedTrails = localStorage.getItem('trails');
+    const savedHydrants = localStorage.getItem('hydrants');
+    let trails = savedTrails ? JSON.parse(savedTrails) : {};
+    let hydrants = savedHydrants ? JSON.parse(savedHydrants) : {};
+
+    let trails = Immutable.fromJS(trails)
+    let hydrants = Immutable.fromJS(hydrants)
+    */
+
     this.state = {
       mode: 'trails',
       selectedTrail: null,
+      canCreate: false,
       trails: Immutable.Map(),
-      hydrants: Immutable.Map()
-    }
-  }
-
-  componentDidMount() {
-    const savedTrails = localStorage.getItem('trails');
-    const savedHydrants = localStorage.getItem('hydrants');
-    const trails = savedTrails ? JSON.parse(savedTrails) : {};
-    const hydrants = savedHydrants ? JSON.parse(savedHydrants) : {};
-    this.setState({
-      trails: Immutable.fromJS(trails),
-      hydrants: Immutable.fromJS(hydrants)
-    });
+      hydrants: Immutable.Map(),
+    };
   }
 
-  componentDidUpdate() {
-    const {trails, hydrants} = this.state;
+  componentDidUpdate(prevProps, prevState) {
+    const { trails, hydrants, selectedTrail } = this.state;
+    /*
     localStorage.setItem('trails', JSON.stringify(trails.toJS()));
     localStorage.setItem('hydrants', JSON.stringify(hydrants.toJS()));
-  }
-
-  modifyTrail = (trailId, editedFields, shouldDelete=false) => {
-    const {trails} = this.state;
-    if (shouldDelete) {
-      this.setState({trails: trails.delete(trailId)});
-    } else {
-      let newTrail = trails.get(trailId).merge(editedFields);
-      this.setState({trails: trails.set(trailId, newTrail)});
+    */
+    if (prevState.selectedTrail !== selectedTrail) {
+      if (prevState.selectedTrail) {
+        const feature = trails.getIn([prevState.selectedTrail, 'feature']);
+        feature.unset('selected');
```

Comment:
Looks like feature is undefined at this point if a trail is selected and deleted. I think we need to check if selectedTrail is null here

Reply:
Ah, nice catch. I missed that one.